### PR TITLE
HRIN-411: Added module files

### PR DESCRIPTION
**Pull request by martinyde**
https://jira.itkdev.dk/browse/HRIN-411

### CODE REVIEW

File: assets/css/dawa.css [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+.dawa-autocomplete-suggestions .dawa-autocomplete-suggestion:last-child {
+  border-bottom-left-radius: inherit;
+  border-bottom-right-radius: inherit;
+  border-bottom-width: 0.0625em;
+}
+
+.dawa-autocomplete-suggestions .dawa-autocomplete-suggestion.dawa-selected,
+.dawa-autocomplete-suggestions .dawa-autocomplete-suggestion:hover {
+  background: #f0f0f0;
+}
```
Comment (rimi-itk) [inline assets/css/dawa.css]: I think all this can be replaced with `@import '~dawa-autocomplete2/css/dawa-autocomplete2.css';`

File: README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# ITK Pretix
+
```
Comment (rimi-itk) [inline README.md:2]: A little more information on what this cool module actually does might be useful here.

File: src/Form/PretixConfigForm.php [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $config = $this->config('itk_pretix.pretixconfig');
+    $form['api_token'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('API Token'),
+      '#maxlength' => 64,
+      '#size' => 64,
+      '#default_value' => $config->get('api_token'),
+    ];
+    return parent::buildForm($form, $form_state);
```
Comment (rimi-itk) [inline src/Form/PretixConfigForm.php:41]: We need some more settings here, e.g. `pretix url` and `organizer (short form)`.

File: src/Plugin/Field/FieldType/PretixFieldType.php [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+              '@max' => $max_length
+            ]),
+          ],
+        ],
+      ]);
+    }
+
+    return $constraints;
+  }
+   */
```
Comment (rimi-itk) [inline src/Plugin/Field/FieldType/PretixFieldType.php]: Remove this (code in comment).

**Review by rimi-itk — CHANGES_REQUESTED** [review]
Looks good, but a few comments have been added.

File: src/Plugin/Field/FieldType/PretixFieldType.php [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\Core\TypedData\DataDefinition;
+use Drupal\Core\TypedData\TypedDataInterface;
+
+/**
+ * Plugin implementation of the 'pretix_field_type' field type.
+ *
+ * @FieldType(
+ *   id = "pretix_field_type",
```
Comment (rimi-itk) [inline src/Plugin/Field/FieldType/PretixFieldType.php:16]: `pretix_field_type` is a very generic name. Maybe `pretix_dates` (or something else) may be a better name.

Comment (rimi-itk) [inline src/Plugin/Field/FieldType/PretixFieldType.php:16]: `pretix_field_type` is a very generic name. Maybe `pretix_dates` (or something else) may be a better name.

**Comment by rimi-itk** [comment]
@martinyde A general question: should this module also provide a field (type) for generic pretix settings related to a piece of content using the “Pretix field type”, e.g. “Event template” and “Update event in pretix”?

File: src/Form/PretixConfigForm.php (hunk 3 above)
Comment (martinyde) [reply src/Form/PretixConfigForm.php:41]: I wasn't quite sure what values you needed so i left that part for you

File: assets/css/dawa.css (hunk 1 above)
Comment (martinyde) [reply assets/css/dawa.css]: Changed css source

File: README.md (hunk 2 above)
Comment (martinyde) [reply README.md:2]: Added A little more information on what this cool module actually does

File: src/Plugin/Field/FieldType/PretixFieldType.php (hunk 4 above)
Comment (martinyde) [reply src/Plugin/Field/FieldType/PretixFieldType.php]: Code removed

**Comment by martinyde** [comment]
@rimi-itk About https://github.com/itk-dev/itk_pretix_d8/pull/1#issuecomment-588280941

I suppose that depends on the time we have...
I'm unsure what use case would benefit from this.

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**